src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_skadi.cpp

```diff
@@ -15,474 +15,931 @@
+
+            void FilterTargets(std::list<WorldObject*>& targets)
             {
-                case 0:
-                    iStart = 8;
-                    iEnd = 37;
-                    break;
-                case 1:
-                    iStart = 38;
-                    iEnd = 68;
-                    break;
-                default:
-                    break;
+                if (targets.size() >= 2)
+                    targets.remove_if([](WorldObject* obj) { return obj->GetEntry() != NPC_GRAUF; });
+            }
+
+            void HandleDamageCalc()
+            {
+                if (Unit* target = GetHitUnit())
+                    SetHitDamage(target->CountPctFromMaxHealth(35));
             }
-            for (uint32 i = iStart; i < iEnd; ++i)
-                me->SummonCreature(NPC_TRIGGER, Location[i]);
+
+            void Register() override
+            {
+                OnObjectAreaTargetSelect += SpellObjectAreaTargetSelectFn(spell_skadi_launch_harpoon_SpellScript::FilterTargets, EFFECT_0, TARGET_UNIT_CONE_ENTRY);
+                OnHit += SpellHitFn(spell_skadi_launch_harpoon_SpellScript::HandleDamageCalc);
+            }
+        };
+
+        SpellScript* GetSpellScript() const override
+        {
+            return new spell_skadi_launch_harpoon_SpellScript();
         }
-    };
+};
+
+class spell_skadi_poisoned_spear : public SpellScriptLoader
+{
+    public:
+        spell_skadi_poisoned_spear() : SpellScriptLoader("spell_skadi_poisoned_spear") { }
+
+        class spell_skadi_poisoned_spear_left_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_skadi_poisoned_spear_left_SpellScript);
+
+            bool Validate(SpellInfo const* /*spell*/) override
+            {
+                if (!sSpellMgr->GetSpellInfo(SPELL_POISONED_SPEAR_PERIODIC))
+                    return false;
+                return true;
+            }
+
+            void HandleScript(SpellEffIndex /*effIndex*/)
+            {
+                GetHitUnit()->CastSpell(GetHitUnit(), SPELL_POISONED_SPEAR_PERIODIC, true);
+            }
 
+            void Register() override
+            {
+                OnEffectHitTarget += SpellEffectFn(spell_skadi_poisoned_spear_left_SpellScript::HandleScript, EFFECT_0, SPELL_EFFECT_SCHOOL_DAMAGE);
+            }
+        };
+
+        SpellScript* GetSpellScript() const override
+        {
+            return new spell_skadi_poisoned_spear_left_SpellScript();
+        }
+};
+
+class spell_summon_gauntlet_mobs_periodic : public SpellScriptLoader
+{
+    public:
+        spell_summon_gauntlet_mobs_periodic() : SpellScriptLoader("spell_summon_gauntlet_mobs_periodic") { }
+
+        class spell_summon_gauntlet_mobs_periodic_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_summon_gauntlet_mobs_periodic_AuraScript);
+
+            void CastTheNextTwoSpells()
+            {
+                uint32 spellId = 0;
```

Comment:
no need to declare spellId outside of the loop and initialize it, it looks weird. please declare it at like 850 where you are assigning it a value already.